### PR TITLE
Heartbeat Fix for Long Asynchronous Tasks

### PR DESCRIPTION
**Pull request by ric-evans**
as seen in https://github.com/icecube/skymap_scanner/actions/runs/6151584209/job/16692007879?pr=199, when the tasks are long, and delayed/heterogeneous/async the heartbeat miss may happen when waiting for new messages rather than during task processing. IOW, we cannot assume the pilot is always saturated

Housekeeping is also expanded to support future situations.

The heartbeat interval is decreased to 5 sec which will have no impact on the broker, but will free up some networking / log space

### CODE REVIEW

File: ewms_pilot/pilot.py [hunk 1]
```diff
@@ -438,91 +480,106 @@ async def _consume_and_reply(
...
-                            [pub.pub] + list(sub._subs.keys())  # type: ignore[arg-type]
-                            if in_queue._broker_client.NAME.lower() == "rabbitmq"
-                            else None
-                        ),
+            if len(pending) >= multitasking:
+                LOGGER.info("At max task concurrency limit")
+            else:
+                LOGGER.info("Listening for incoming message...")
+                try:
+                    in_msg = await anext(sub.iter_messages())  # -> in_queue.timeout
```
Comment (dsschult) [inline ewms_pilot/pilot.py:527]: I'm not sure what the insides of `sub.iter_messages()` looks like, but it's odd that we're creating the iterator on every loop.  Maybe it's fine?

**Review by dsschult — COMMENTED** [review]
I think things are mostly okay, though some of the code is getting hard to follow.  This file might be getting a little long, and could be split.

File: ewms_pilot/pilot.py (hunk 1 above)
Comment (ric-evans) [reply ewms_pilot/pilot.py:527]: it's technically okay, but it is odd-looking. I could add a non-iterator function to mqclient. Otherwise, I could make an iterator instance and re-instantiate that on `StopAsyncIteration`. Thoughts?

Comment (dsschult) [reply ewms_pilot/pilot.py:527]: The typical pattern is to save the iterator to a variable outside the loop and call `anext(iter)` as needed.  But not sure if there are advantages to creating the iterator again.

Comment (ric-evans) [reply ewms_pilot/pilot.py]: once `StopAsyncIteration` is raised, the iterator is exhausted so each future call will raise `StopAsyncIteration` again. Recreating the instance restarts the internals

**Comment by ric-evans** [comment]
> This file might be getting a little long, and could be split.

I agree. I'll split the file post-merge. That way, we can preserve line history.
